### PR TITLE
docs(vrl): document punycode as an IDN function

### PR DESCRIPTION
**Pull request by esensar**
This adds a note to `encode_punycode` and `decode_punycode` to inform users that it is meant to be used in IDN context and that passed values are handled like domain names. This is important because punycode by default doesn't use `xn--` prefix, but it is added in IDN context.

Related: https://github.com/vectordotdev/vrl/pull/1104

### CODE REVIEW

**Review by pront — APPROVED** [review]

File: website/cue/reference/remap/functions/decode_punycode.cue [hunk 1]
```diff
@@ -3,7 +3,7 @@ package metadata
 remap: functions: decode_punycode: {
 	category:    "Codec"
 	description: """
-		Decodes a [punycode](\(urls.punycode)) encoded `value`, like an internationalized domain name ([IDN](\(urls.idn))).
+		Decodes a [punycode](\(urls.punycode)) encoded `value`, like an internationalized domain name ([IDN](\(urls.idn))). This function assumes that the value passed is meant to be used in IDN context and that it is either a domain name or a part of it.
```
Comment (aliciascott) [inline website/cue/reference/remap/functions/decode_punycode.cue:6]:
```suggestion
		Decodes a [punycode](\(urls.punycode)) encoded `value`, such as an internationalized domain name ([IDN](\(urls.idn))). This function assumes that the value passed is meant to be used in IDN context and that it is either a domain name or a part of it.
```

**Review by aliciascott — APPROVED** [review]
Hi @esensar just a small suggestion for wording but approved

File: website/cue/reference/remap/functions/decode_punycode.cue (hunk 1 above)
Comment (esensar) [reply website/cue/reference/remap/functions/decode_punycode.cue]: Thank you!

**Comment by esensar** [comment]
I just saw this has failed workflow. This doesn't need changelog, right? Since the change is in VRL

**Pull request merged**